internal/apiserver/clusterservice/clusterimpl.go

```diff
@@ -2063,6 +2082,102 @@ func isMissingS3Config(request *msgs.CreateClusterRequest) bool {
+// to create a new cluster
+func validateDataSourceParms(request *msgs.CreateClusterRequest) error {
+
+	namespace := request.Namespace
+	restoreClusterName := request.PGDataSource.RestoreFrom
+	restoreOpts := request.PGDataSource.RestoreOpts
+
+	if restoreClusterName == "" && restoreOpts == "" {
+		return nil
+	}
+
+	// first verify that a "restore from" parameter was specified if the restore options
+	// are not empty
+	if restoreOpts != "" && restoreClusterName == "" {
+		return fmt.Errorf("A cluster to restore from must be specified when providing restore " +
+			"options")
+	}
+
+	// next verify whether or not a PVC exists for the cluster we are restoring from
+	if _, err := apiserver.Clientset.CoreV1().PersistentVolumeClaims(namespace).Get(
+		fmt.Sprintf(util.BackrestRepoPVCName, restoreClusterName),
+		metav1.GetOptions{}); err != nil {
+		return fmt.Errorf("Unable to find PVC %s for cluster %s, cannot to restore from the "+
+			"specified data source", fmt.Sprintf(util.BackrestRepoPVCName, restoreClusterName),
+			restoreClusterName)
+	}
+
+	// now verify that a pgBackRest repo secret exists for the cluster we are restoring from
+	backrestRepoSecret, err := apiserver.Clientset.CoreV1().Secrets(namespace).Get(
+		fmt.Sprintf(util.BackrestRepoSecretName, restoreClusterName), metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Unable to find secret %s for cluster %s, cannot restore from the "+
+			"specified data source",
+			fmt.Sprintf(util.BackrestRepoSecretName, restoreClusterName), restoreClusterName)
+	}
+
+	// next perform general validation of the restore options
+	if err := backupoptions.ValidateBackupOpts(restoreOpts, request); err != nil {
+		return fmt.Errorf("%s: %w", ErrInvalidDataSource, err)
+	}
+
+	// now detect if an 's3' repo type was specified via the restore opts, and if so verify that s3
+	// settings are present in backrest repo secret for the backup being restored from
+	s3Restore, err := backrest.S3RepoTypeCLIOptionExists(restoreOpts)
+	if err != nil {
+		return fmt.Errorf("%s: %w", ErrInvalidDataSource, err)
+	}
+	if s3Restore && isMissingExistingDataSourceS3Config(backrestRepoSecret) {
+		return fmt.Errorf("Secret %s is missing the S3 configuration required to restore "+
+			"from an S3 repository", backrestRepoSecret.GetName())
+	}
+
+	// finally, verify that the cluster being restored from is in the proper status, and that no
+	// other clusters currently being bootstrapping from the same cluster
+	clusterList := &crv1.PgclusterList{}
+	if err := kubeapi.Getpgclusters(apiserver.RESTClient, clusterList, namespace); err != nil {
+		return fmt.Errorf("%s: %w", ErrInvalidDataSource, err)
+	}
+	for _, cl := range clusterList.Items {
+
+		if cl.GetName() == restoreClusterName &&
+			cl.Status.State == crv1.PgclusterStateShutdown {
+			return fmt.Errorf("Unable to restore from cluster %s because it has a %s "+
+				"status", restoreClusterName, string(cl.Status.State))
+		}
```

Comment:
The pgBackRest repository won't be running.  We could start the repo only for the restore only, but I felt that as long as the cluster still exists, it alone should control when it's repository running (i.e. it was likely shutdown for a reason, and should stay that way until it decides otherwise).  So basically only allowing a new cluster to control another repo when there is no longer a specific PG cluster managing/using it.